client/my-sites/media-library/test/list.jsx

```diff
@@ -3,209 +3,287 @@
+const withSelectedItems = ( Component ) => ( props ) => {
+	const [ selectedItems, setSelectedItems ] = useState( [] );
+
+	selectMediaItems.mockImplementation( ( siteId, mediaItems ) => {
+		setSelectedItems( mediaItems );
+	} );
+
+	return (
+		<Component { ...props } selectedItems={ selectedItems } selectMediaItems={ selectMediaItems } />
+	);
+};
 
 describe( 'MediaLibraryList item selection', () => {
-	let wrapper;
-	let mediaList;
-
-	const selectMediaItems = jest.fn();
-
-	function toggleItem( itemIndex, shiftClick ) {
-		mediaList.toggleItem( fixtures.media[ itemIndex ], shiftClick );
-	}
-
-	function expectSelectedItems( ...args ) {
-		defer( function () {
-			expect( mockSelectedItems ).toContain(
-				args.map( function ( arg ) {
-					return fixtures.media[ arg ];
-				} )
-			);
-		} );
-	}
+	let originalScrollTo;
 
-	beforeEach( () => {
-		mockSelectedItems.length = 0;
+	beforeAll( () => {
+		originalScrollTo = window.scrollTo;
+		window.scrollTo = () => null;
+	} );
+
+	afterAll( () => {
+		window.scrollTo = originalScrollTo;
+	} );
+
+	afterEach( () => {
+		selectMediaItems.mockClear();
 	} );
 
 	describe( 'multiple selection', () => {
-		beforeEach( () => {
-			wrapper = mount(
-				<MediaList
-					filterRequiresUpgrade={ false }
-					site={ { ID: DUMMY_SITE_ID } }
-					media={ fixtures.media }
-					mediaScale={ 0.24 }
-					moment={ moment }
-					selectedItems={ [] }
-					selectMediaItems={ selectMediaItems }
-				/>
-			);
-			mediaList = wrapper.find( MediaList ).instance();
-		} );
+		const props = {
+			filterRequiresUpgrade: false,
+			media,
+			mediaScale: 0.24,
+			moment,
+			selectMediaItems,
+			selectedItems: [],
+			site: { ID: DUMMY_SITE_ID },
+		};
 
-		afterEach( () => {
-			wrapper.unmount();
-		} );
+		test( 'allows selecting single items', async () => {
+			const user = userEvent.setup();
+			render( <MediaList { ...props } /> );
+			const [ item1, item2 ] = media;
+
+			await user.click( screen.getByTestId( `list-item-${ item1.ID }` ) );
+			expect( selectMediaItems ).toHaveBeenCalledWith( DUMMY_SITE_ID, [ item1 ] );
 
-		test( 'allows selecting single items', () => {
-			toggleItem( 0 );
-			expectSelectedItems( 0 );
-			toggleItem( 2 );
-			expectSelectedItems( 0, 2 );
+			await user.click( screen.getByTestId( `list-item-${ item2.ID }` ) );
+			expect( selectMediaItems ).toHaveBeenCalledWith( DUMMY_SITE_ID, [ item2 ] );
 		} );
 
-		test( 'allows selecting multiple items at once by Shift+clicking', () => {
-			toggleItem( 0 );
-			toggleItem( 4, true ); // Shift+click to select items 0 through 4
-			expectSelectedItems( 0, 1, 2, 3, 4 );
+		test( 'allows selecting multiple items at once by Shift+clicking', async () => {
+			const user = userEvent.setup();
+			render( <MediaList { ...props } /> );
+
+			await user.click( screen.getByTestId( `list-item-${ media[ 1 ].ID }` ) );
+			await user.keyboard( '{Shift>}' ); // `>` means keep the `Shift` key pressed
```

Comment:
Would it make the test more thorough if at the end we release the shift button and assert that nothing additional has changed?

Same for the rest of the tests where we keep a key pressed without releasing it.

Reply:
That's actually a fair point. I'll make some adjustments 👍🏻

Reply:
Added https://github.com/Automattic/wp-calypso/pull/69375/commits/f7904f089bdaae14e104764271f064ae891b81c0 with which we're releasing the `Shift` key before the last assertion.